storm-core/src/clj/backtype/storm/daemon/supervisor.clj

```diff
@@ -349,6 +350,20 @@
                         (keys keepers))
            (zipmap (vals new-worker-ids) (keys new-worker-ids))
            ))
+
+    ;; check storm topology code dir exists before launching workers
+    (doseq [[port assignment] reassign-executors]
+      (let [storm-cluster-state (:storm-cluster-state supervisor)
+            downloaded-storm-ids (set (read-downloaded-storm-ids conf))
+            storm-id (:storm-id assignment)
+            assignment-info (.assignment-info-with-version storm-cluster-state storm-id nil)
```

Comment:
This is going to cause a large load on zookeeper.  There is a reason we have the cache for the assignment info, and we only download it when it has changed.  We really should use the cache here if we can.